util/Order.cpp

```diff
@@ -1304,3 +1304,35 @@ bool GiveObjectToEmpireOrder::UndoImpl() const {
     }
     return false;
 }
+
+////////////////////////////////////////////////
+// ForgetOrder
+////////////////////////////////////////////////
+ForgetOrder::ForgetOrder() :
+    Order(),
+    m_object_id(INVALID_OBJECT_ID)
+{}
+
+ForgetOrder::ForgetOrder(int empire, int object_id) :
+    Order(empire),
+    m_object_id(object_id)
+{}
+
+void ForgetOrder::ExecuteImpl() const {
+    ValidateEmpireID();
+    int empire_id = EmpireID();
+    TemporaryPtr<Fleet> fleet = GetFleet(m_object_id);
```

Comment:
Do not look behind the curtain.  We are making sausage here.

I started from and only tested it with the ghost fleets.  It is fixed now.